demos/app/iot_demo.c

```diff
@@ -118,19 +118,55 @@ int main( int argc,
     /* For a secured connection, set the members of the credentials. */
     if( demoArguments.securedConnection == true )
     {
-        /* Set credential paths. */
+        /* Set credential information. */
         credentials.pClientCert = demoArguments.pClientCertPath;
         credentials.pPrivateKey = demoArguments.pPrivateKeyPath;
         credentials.pRootCa = demoArguments.pRootCaPath;
+        
+        /* Set the MQTT username, as long as it's not empty or NULL. */
+        if( demoArguments.pUserName != NULL )
+        {
+            credentials.userNameSize = strlen( demoArguments.pUserName );
+            if( credentials.userNameSize > 0 )
```

Comment:
Do we need to enforce a maximum size for `credentials.userNameSize`?

Reply:
Good catch, thank you, fix is pending.

Reply:
Fixed.